proxy/apifilterproxy/filters/http/generic_http_filter.go

```diff
@@ -0,0 +1,75 @@
+package http
+
+import (
+	"bytes"
+	"encoding/json"
+	log "github.com/Sirupsen/logrus"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/util"
+)
+
+const (
+	name = "http"
+)
+
+func init() {
+	httpFilter := &GenericHTTPFilter{}
+	if err := filters.RegisterAPIFilter(name, httpFilter); err != nil {
+		log.Fatalf("Could not register %s filter", name)
+	}
+
+	log.Infof("Configured %s API filter", httpFilter.GetName())
+
+}
+
+type GenericHTTPFilter struct {
+}
+
+func (*GenericHTTPFilter) GetName() string {
+	return name
+}
+
+func (f *GenericHTTPFilter) ProcessFilter(filter model.FilterData, input model.APIRequestData) (model.APIRequestData, error) {
+	output := model.APIRequestData{}
+	bodyContent, err := json.Marshal(input)
+	if err != nil {
+		return output, err
+	}
+
+	log.Debugf("Request => " + string(bodyContent))
```

Comment:
the correct way to write this would be log.Debugf("Request => %s", bodyContent)
that way, the type conversion (which involves a potentially expensive copy) only occurs when debug is on (bc the params are passed to Debugf which checks to see if debug is on before doing the conversion)